packages/react/src/ImageNew.tsx

```diff
@@ -0,0 +1,379 @@
+      src: loader(
+        src,
+        placeholderWidth,
+        aspectRatio && placeholderWidth
+          ? placeholderWidth * (parseAspectRatio(aspectRatio) ?? 1)
+          : undefined
+      ),
+      alt,
+      sizes,
+      style: {
+        width: normalizedWidth,
+        height: normalizedHeight,
+        aspectRatio,
+      },
+      loading,
+      ...passthroughProps,
+    });
+  }
+}
+
+function unitsMatch(
+  width: string | number = '100%',
+  height: string | number = 'auto'
+) {
+  return (
+    getUnitValueParts(width.toString()).unit ===
+    getUnitValueParts(height.toString()).unit
+  );
+  /*
+      Given:
+        width = '100px'
+        height = 'auto'
+      Returns:
+        false
+
+      Given:
+        width = '100px'
+        height = '50px'
+      Returns:
+        true
+   */
+}
+
+function getUnitValueParts(value: string) {
+  const unit = value.replace(/[0-9.]/g, '');
+  const number = parseFloat(value.replace(unit, ''));
+
+  return {
+    unit: unit === '' ? (number === undefined ? 'auto' : 'px') : unit,
+    number,
+  };
+  /*
+      Given:
+        value = '100px'
+      Returns:
+        {
+          unit: 'px',
+          number: 100
+        }
+   */
+}
+
+function getNormalizedFixedUnit(value?: string | number) {
+  if (value === undefined) {
+    return;
+  }
+
+  const {unit, number} = getUnitValueParts(value.toString());
+
+  switch (unit) {
+    case 'em':
+      return number * 16;
+    case 'rem':
+      return number * 16;
+    case 'px':
+      return number;
+    case '':
+      return number;
+    default:
+      return;
+  }
+  /*
+      Given:
+        value = 16px | 1rem | 1em | 16
+      Returns:
+        16
+
+      Given:
+        value = 100%
+      Returns:
+        undefined
+   */
+}
+
+function isFixedWidth(width: string | number) {
+  const fixedEndings = new RegExp('px|em|rem', 'g');
+  return (
+    typeof width === 'number' ||
+    (typeof width === 'string' && fixedEndings.test(width))
+  );
+  /*
+    Given:
+      width = 100 | '100px' | '100em' | '100rem'
+    Returns:
+      true
+  */
+}
+
+export function generateShopifySrcSet(
+  src: string,
+  sizesArray?: Array<{width?: number; height?: number; crop?: Crop}>
+) {
+  if (sizesArray?.length === 0 || !sizesArray) {
+    return src;
+  }
+
+  return sizesArray
+    .map(
+      (size) =>
+        shopifyLoader(src, size.width, size.height, size.crop) +
+        ' ' +
+        size.width +
+        'w'
+    )
+    .join(`, `);
+  /*
+      Given:
+        src = 'https://cdn.shopify.com/static/sample-images/garnished.jpeg'
+        sizesArray = [
+          {width: 200, height: 200, crop: 'center'},
+          {width: 400, height: 400, crop: 'center'},
+        ]
+      Returns:
+        'https://cdn.shopify.com/static/sample-images/garnished.jpeg?width=200&height=200&crop=center 200w, https://cdn.shopify.com/static/sample-images/garnished.jpeg?width=400&height=400&crop=center 400w'
```

Comment:
I've just updated the loader to only accept a single argument (a params object) so this can be more flexible. I updated the `Crop` type to accept an object with the crop region params above.

So… 

```ts
type Crop =
  | 'center'
  | 'top'
  | 'bottom'
  | 'left'
  | 'right'
  | {top?: number; left?: number; width?: number; height?: number}
  | undefined;
```

Does that look right?